### ISSUE TITLE
Shield health for explosions

### ISSUE THREAD

**Issue by sed-and-awk** [LemADEC/WarpDrive#452]
**Is your feature request related to a problem? Please describe.**
When using ICBM missiles against shields only the EMP missile has any effect

**Describe the solution you'd like**
Add some sort of health system for the shields that is not just blast resistance so that multiple explosions in a row could break through such as the hull currently has against lasers

 **Describe alternatives you've considered**
Have all damage to the shield drain power

**Comment by LemADEC** [comment]
Already implemented since the origin of the mod. Enable logging to see the numbers.

**Comment by sed-and-awk** [comment]
Oh then maybe increase damage taken because a nuke did practically nothing

**Comment by LemADEC** [comment]
No, instantly breaking a force field makes no sense in terms of gameplay.
A force field is there to protect the base overtime and recharge itself, it's not there to be instantly broken by a single block.

**Comment by lellasone** [comment]
I tried decreasing the "projector_explosion_scale=N" by a factor of 1000, and I haven't been able to break a shield fed by 3k RF/T (three basic energy cells) using ICBM explosives even in creative. Do you have a sense for how many nukes/missiles/ect it should take to break a forcefield? 

I have the logging enabled, but I don't really know how to interpret the messages.

**Comment by LemADEC** [comment]
Force field impact depends on:
- the number of force field blocks hit
- for each block, its square distance from the explosion source
- the explosion strength (i.e. it's generally its radius)
- the active upgrades on the field (some upgrades can drastically affect the sensitivity)
In the logs, the magnitude is the distance from the explosion source (i.e. the magnitude of the relative position).
So, in first approximation, energy impact for each force field block in range is `strength / (magnitude * magnitude * exposion_scale)`.

In other words, similarly to real life, stronger explosive are more effective when detonating away from the target, not directly on it, but not too far either. Also, especially with IC2, you can have a compression/focalization effect by enclosing the explosive.
Obviously, in general, you want to target the center of a surface to maximize coverage, and by extension, a sphere is better at absorbing damage provided it comes with the thickness upgrade.

**Comment by lellasone** [comment]
That's a really cool system, thank you for explaining it.  I really like the way it plays out using IC2 nukes, the way that breaching missiles seem to work (where the delete a few shield blocks, but only a few) is a pretty nice mechanic too. Really encourages the use of speed modules. 

With a bit more testing my issue seems to be that a lot of ICBM explosives don't really interact properly with Warp Drive shield or hull blocks. The basic missiles work, but don't do enough damage to matter (nor should they), while the higher tier explosives like nukes or thermobaric missiles get ignored entirely (for hull blocks they do no damage, and for shields they trigger the force field log message, but not the subsequent weapon log message).
